tests/client_tests/integration/test_reconnect.py

```diff
@@ -38,4 +39,17 @@ class ReconnectTests(IntegrationTestCase):
         self.ledger.config['connect_timeout'] = 1
         self.ledger.config['default_servers'] = [('10.0.0.1', 12)] + list(conf['default_servers'])
         await self.ledger.start()
-        self.assertTrue(self.ledger.network.is_connected)
\ No newline at end of file
+        self.assertTrue(self.ledger.network.is_connected)
+
+    async def test_pick_fastest(self):
+        # local server that is listening but wont reply
+        proto = RPCSession()
+        proto.handle_request = lambda _: asyncio.sleep(10)
+        server = await self.loop.create_server(lambda: proto, host='127.0.0.1', port=1337)
+        await self.ledger.stop()
+        conf = self.ledger.config
+        self.ledger.config['default_servers'] = [('127.0.0.1', 1337)] + list(conf['default_servers'])
+        self.ledger.config['connect_timeout'] = 30
+        await asyncio.wait_for(self.ledger.start(), timeout=1)
+        self.assertTrue(self.ledger.network.is_connected)
+        self.assertEqual(self.ledger.network.client.server, conf['default_servers'][-1])
```

Comment:
Will this test still pass without the feature? I think it will still pass: it'll fail to connect to the first server and then successfully connect to the real one.

I'd suggest making three fake RPC sessions and not use the real one at all. For the 3 fake ones have them respond with second delays: 0.5, 0.3, 0.7. and then check that the middle one wins.